tests/e2e/forms/submit-default-delivery-form.wdio-spec.js

```diff
@@ -0,0 +1,374 @@
+    }
+
+    // Verify alive babies UUIDs are unique
+    const aliveBabyUUIds = [];
+    for (let i = 0; i < noOfAliveBabies; i++) {
+      aliveBabyUUIds.push(await deliveryReport.getAliveBabyUUID(i));
+    }
+
+    expect(deadBabyUUIds.length).to.equal(noOfDeadBabies);
+    expect(aliveBabyUUIds.length).to.deep.equal(noOfAliveBabies);
+    expect(_.uniq(deadBabyUUIds).length).to.equal(noOfDeadBabies);
+    expect(_.uniq(aliveBabyUUIds).length).to.deep.equal(noOfAliveBabies);
+
+    await reportsPage.editReport(reportId);
+    await genericForm.nextPage();
+    await genericForm.nextPage();
+    await genericForm.nextPage();
+    await genericForm.nextPage();
+    await genericForm.nextPage();
+    await genericForm.nextPage();
+    await genericForm.nextPage();
+    await genericForm.nextPage();
+    await reportsPage.submitForm();
+    await sentinelUtils.waitForSentinel();
+
+    const updatedReport = await utils.getDoc(reportId);
+    const exclude = [
+      'instanceID',
+      'baby_death_profile_doc', // every extra doc is duplicated when editing
+      'child_doc', // every extra doc is duplicated when editing
+    ];
+    // fields remain identical
+    expect(updatedReport.hidden_fields).to.deep.equal(initialReport.hidden_fields);
+    expect(updatedReport.fields).excludingEvery(exclude).to.deep.equal(initialReport.fields);
+    // input fields are filled as expected
+    expect(initialReport.fields.inputs).excludingEvery(['meta', 'patient_id']).to.deep.equal({
+      source: 'user',
+      source_id: '',
+      user: {
+        contact_id: userData.userContactDoc._id,
+        name: userData.userContactDoc.name,
+        phone: userData.userContactDoc.phone,
+      },
+      contact: {
+        _id: userData.userContactDoc._id,
+        name: userData.userContactDoc.name,
+        short_name: '',
+        date_of_birth: '',
+        sex: '',
+        parent: {
+          _id: userData.userContactDoc.parent._id,
+          parent: { _id: '', parent: { _id: '' }, contact: { chw_name: '', phone: '' } },
+        }
+      }
+    });
+
+    // created extra docs are identical
+    const intialDeadBabies = await utils.getDocs(deadBabyUUIds);
+    const initialAliveBabies = await utils.getDocs(aliveBabyUUIds);
+
+    // Verify dead babies UUIDs are unique
+    const updatedDeadBabyUUIds = [];
+    for (let i = 0; i < noOfDeadBabies; i++) {
+      updatedDeadBabyUUIds.push(await deliveryReport.getDeadBabyUUID(i));
+    }
+
+    // Verify alive babies UUIDs are unique
+    const updatedAliveBabyUUIds = [];
+    for (let i = 0; i < noOfAliveBabies; i++) {
+      updatedAliveBabyUUIds.push(await deliveryReport.getAliveBabyUUID(i));
+    }
+
+    expect(updatedDeadBabyUUIds.length).to.equal(noOfDeadBabies);
+    expect(updatedAliveBabyUUIds.length).to.deep.equal(noOfAliveBabies);
+    expect(_.uniq(updatedDeadBabyUUIds).length).to.equal(noOfDeadBabies);
+    expect(_.uniq(updatedAliveBabyUUIds).length).to.deep.equal(noOfAliveBabies);
```

Comment:
Would be nice to also assert that the initialAlive/DeadBabies and updatedAlive/DeadBabies are disjoint sets (so that when we start comparing baby contacts, we know that we are not just comparing the same contact with itself).